pages/blog/introduction-to-glee-a-spec-first-framework.md

```diff
@@ -0,0 +1,169 @@
+
+ - Glee simplifies the process of creating and maintaining connections, allowing you to concentrate solely on developing code that meets your business needs. By handling performance, scalability, resilience, and all other aspects necessary for production-readiness, Glee frees you from the burden of managing these technicalities, enabling you to focus on delivering a high-quality application that meets the demands of your users.
+
+
+# Getting Started With Glee
+
+Let's create a simple WebSocket API using glee to understand its magic. We will create a simple WebSocket server that receives a current time from the client and then send a "good morning", "good evening" or "good night" respectively.
+
+### Setting up a glee project.
+
+To work with Glee, you must install NPM and NodeJs version 10 or higher. To check if you have both installed, run the following commands in your terminal. 
+
+```bash
+# check if node is installed
+node -v
+# or
+node --version
+
+# check if NPM is installed
+npm -v
+# or
+npm --version
+
+```
+
+If you don't have any of the above tools missing go ahead and install them.
+
+
+
+#### Create glee project 
+
+We recommend creating a new Glee app using our official [CLI](https://github.com/asyncapi/cli) which sets up everything automatically. (You don't need to create an empty directory. create-glee-app will make one for you.) To create a project, run:
+
+```bash
+
+asyncapi new glee 
+
+```
+
+> To install AsyncAPI CLI either use [npm](https://www.npmjs.com/package/@asyncapi/cli) or install binaries for your operating system from https://github.com/asyncapi/cli/releases 
+
+
+Once the process is complete you should have a new Glee app ready for development and see these files that were made.
+
+<center>
+
+![Glee File Structure](/img/posts/glee-introduction/glee-file-structure.webp)
+
+</center>
+
+
+
+#### Define our Spec for our API.
+
+Glee being a spec-first framework, development starts with defining your API spec. For our case we will define our API:
+
+
+```yaml:asyncapi.yaml
+asyncapi: 2.1.0
+info:
+  title: Greet Bot
+  version: 0.1.0
+servers:
+  websockets:
+    url: ws://0.0.0.0:3000
+    protocol: ws
+channels:
+  greet:
+    publish:
+      operationId: onGreet
+      message:
+        $ref: '#/components/messages/time'
+    subscribe:
+      message:
+        $ref: '#/components/messages/greet'
+components:
+  messages:
+    time:
+      payload:
+        type: object
+        properties:
+          currentTime:
+            type: number
+          name:
+            type: string
+    greet:
+      payload:
+        type: string
+```
+
+This will be the Specification that defines our API, in our case, it is very simple, as we will be sending a name and the time of the day, and our API will greet us accordingly. 
```

Comment:
```suggestion
This will be the specification that defines our API. In our case, it is very simple, as we will send a name and the time of the day, and our API will greet us accordingly. 
```